### PR TITLE
feat(collection): updating find API

### PR DESCRIPTION
**Pull request by daprahamian**
Find api is now updated and clarified. You can no longer pass in
fields, and you cannot pass in individual options as parameters.

BREAKING CHANGE:
`find` and `findOne` no longer support the `fields` parameter.
You can achieve the same results as the `fields` parameter by
using `Cursor.prototype.project` or by passing the `projection`
property in on the `options` object . Additionally, `find` does not
support individual options like `skip` and `limit` as positional
parameters. You must pass in these parameters in the `options` object

### CODE REVIEW

**Comment by daprahamian** [comment]
Note that I also added a commit that just fixes lint errors. Feel free to ignore that commit in the review

File: lib/collection.js [hunk 1]
```diff
@@ -3269,39 +3174,4 @@ var getReadPreference = function(self, options, db) {
   return options;
 };
 
-var testForFields = {
```
Comment (mbroadst) [inline lib/collection.js]: so very nice to get rid of this :+1:

File: test/functional/command_write_concern_tests.js [hunk 2]
```diff
@@ -718,17 +722,19 @@ describe('Command Write Concern', function() {
...
             var db = client.db(configuration.db);
 
-            db.collection('test').dropIndexes({
-              w: 2,
-              wtimeout: 1000
-            },
-            function(err) {
-              test.equal(null, err);
-              test.deepEqual({ w: 2, wtimeout: 1000 }, commandResult.writeConcern);
+            db.collection('test').dropIndexes(
```
Comment (mbroadst) [inline test/functional/command_write_concern_tests.js]: are these all just whitespace changes? If so, let's try to keep those relegated to separate commits/PRs

File: test/unit/sessions/collection_tests.js [hunk 3]
```diff
@@ -40,9 +40,7 @@ describe('Sessions', function() {
 
           return coll
             .insert({ a: 42 }, { session: session })
-            .then(() =>
-              coll.findOne({}, null, { session: session, readConcern: { level: 'majoroy' } })
-            )
+            .then(() => coll.findOne({}, { session: session, readConcern: { level: 'majoroy' } }))
```
Comment (mbroadst) [inline test/unit/sessions/collection_tests.js]: `level: 'majority'` ?

**Comment by mbroadst** [comment]
I stopped my review short because I want to focus on the idea of supporting `project` in the `options` for `find`.  Lets make that choice first, and then work from there.

**Comment by jlord** [comment]
I think supporting `project` in `options` could be good too. Is there anything comparable in another driver?

> We may want to document the options object for find more clearly

I think we should also add code samples as well 👍

File: lib/collection.js [hunk 4]
```diff
@@ -264,49 +230,28 @@ Collection.prototype.find = function() {
...
-      } else {
-        var l = options.fields.length;
+  let project = options.project || options.fields;
 
-        for (i = 0; i < l; i++) {
-          fields[options.fields[i]] = 1;
-        }
-      }
+  if (project && !Buffer.isBuffer(project) && Array.isArray(project)) {
+    const objectProject = {};
```
Comment (mbroadst) [inline lib/collection.js]: can't we just do this like:
```
project = project.reduce((result, field) => { result[field] = 1; return result; }, {});
if (Object.keys(project).length === 0) project['_id'] = 1;
```

Comment (mbroadst) [inline lib/collection.js]: can't we just do this in two lines? 
```
project = project.reduce((result, field) => { result[field] = 1; return result; }, {});
if (Object.keys(project).length === 0) project['_id'] = 1;
```

File: lib/collection.js [hunk 5]
```diff
@@ -192,57 +192,23 @@ Object.defineProperty(Collection.prototype, 'hint', {
 /**
  * Creates a cursor for a query that can be used to iterate over results from MongoDB
  * @method
- * @param {object} query The cursor query object.
- * @param {Object} [options] Optional settings
+ * @param {object} [query={}] The cursor query object.
+ * @param {object} [options={}] Optional settings
```
Comment (mbroadst) [inline lib/collection.js]: please document the supported options here
